Small fixes in the Readme

0xc0170 opened the pull request:
@c1728p9 @ashok-rao Please review

0xc0170 committed:
Readme: fix import example name

0xc0170 committed:
Readme: add note about two binaries produced

ashok-rao reviewed (approved):
LGTM 👍.. Apart from the mention of the blinky_application.bin in main.cpp in the repo, other docs changes are fine. Will @c1728p9 be addressing the point in main.cpp?

0xc0170 commented:
> Apart from the mention of the blinky_application.bin in main.cpp in the repo, other docs changes are fine. Will @c1728p9 be addressing the point in main.cpp?

This define: https://github.com/ARMmbed/mbed-os-example-bootloader/blob/master/main.cpp#L5 ?

ashok-rao commented:
Yes @0xc0170 , that one.. SD card for just building the bootloader binary (as per the README) seems a bit out-of-place..also we might want to make this as starting point for all examples..even filesystems ones.. so better to keep it generic than hard code a particular binary / example to it..

@c1728p9 @screamerbg , @sg- .. what do you think?

0xc0170 committed:
Readme: add bootloader tutorial reference

0xc0170 committed:
Readme: replace Nucleo target references by <TARGET_NAME>

0xc0170 commented:
> Yes @0xc0170 , that one.. SD card for just building the bootloader binary (as per the README) seems a bit out-of-place..also we might want to make this as starting point for all examples..even filesystems ones.. so better to keep it generic than hard code a particular binary / example to it..

New update - removed not needed parts (CI, SD card), replace nucleo target by a generic target name

ashok-rao commented:
LGTM .. we just need @c1728p9 to confirm about https://github.com/ARMmbed/mbed-os-example-bootloader/blob/master/main.cpp#L5 .. Rest all good to go!

sg- commented:
I've always liked the idea of offset built images having the offset in the name (see DAPLink). I think we should implement that here / in the tools too rather than using the word bootloader everywhere

0xc0170 commented:
> I've always liked the idea of offset built images having the offset in the name (see DAPLink). I think we should implement that here / in the tools too rather than using the word bootloader everywhere

sounds good to me to have it in the name. @c1728p9 What do you think? If we make it in the tools, I can update the docs

ashok-rao commented:
That would avoid a LOT of confusions.. +1 to have the distinctions in the file names itself.

c1728p9 commented on README.md:
```diff
@@ -2,29 +2,29 @@
 ## Required hardware
 * A supported board - [u-blox EVK-ODIN-W2](https://developer.mbed.org/platforms/ublox-EVK-ODIN-W2/), [Nucleo F429ZI](https://developer.mbed.org/platforms/ST-Nucleo-F429ZI/) or [K64F](https://developer.mbed.org/platforms/FRDM-K64F/).
-* CI Test shield.
-* SD card.
```
The bootloader in the example is setup specifically to use an SD card on the CI Test shield. Without this none of the bootloader functionality can be used.

0xc0170 replied:
Will fix

ashok-rao replied:
@c1728p9 , as per the README, this example only "builds" the bootloader binary.. there is no mention of updating it or any other BL functionality. Should we modify the README then?

0xc0170 replied:
I'll fix it

c1728p9 commented:
I don't have strong feelings either way on having the starting address of an application/bootloader as part of the name. @theotherjimmy what are your thoughts on this?

ashok-rao commented:
>I don't have strong feelings either way on having the starting address of an application/bootloader as part of the name. @theotherjimmy what are your thoughts on this?

I think this refers to using the "_application" or the "_bl" in the file names? And not the actual offset addresses..I maybe wrong..

ashok-rao commented on README.md:
```diff
@@ -72,6 +72,8 @@ Total Flash memory (text + data + misc): 60554 bytes
+
 ## Next steps
 
 When the build succeeds, you have created a bootloader for your target. The next step is to build an application you can combine with your bootloader to create a loadable image. You can find a blinky application that uses this bootloader in the [mbed-os-example-bootloader-blinky](https://github.com/ARMmbed/mbed-os-example-bootloader-blinky) project.
```
"When it builds, you have created a bootloader for your target. The next step is to build an application .... " I think this statement is ambiguous. So, this application "only" builds the BL binary. I have to go to another application that uses this bootloader. So, why do we need to mention the SD card or test the "update" functionality of the bootloader here with an  SD card??

0xc0170 replied:
I'll rewrite this section, noticed the same

ashok-rao reviewed (changes requested):
Some lines in the "Next steps" section are confusing. Please review.

How about finding some other means of testing the BL -> like flash the new bootloader, make sure the DAPLink interface can still be upgraded/downgraded on the target or atleast make sure the DAPLink can be written to the device again & the target is still enumerated as mbed ? etc etc..

sg- commented:
The proposal would be the following:
- application without a bootloader, <program name>
- application with constrain, <program name>-<?? or maybe blank ??>
- application with bootloader.bin, <program name>-<loadaddress 0x123455>

0xc0170 committed:
Readme: add SD card binary location

This should explain why SD card is needed and where bootloader should
find the application.

0xc0170 commented:
@ashok-rao I added SD card explanation, please review

0xc0170 commented:
@ashok-rao Please review, or make edits here. I would like to proceed with this Readme update.

ashok-rao commented:
@0xc0170 .. Ok, let's proceed with this for now. I'll make edits as I find suitable and raise other PR's. Thanks!

ashok-rao commented:
I have approved now. Please go ahead to merge :-)
